### ISSUE TITLE
[android]: mark in editor.marks is not inserted on insertText

### ISSUE THREAD

**Issue by imdbsd** [ianstormtaylor/slate#4405]
**Description**
Currently on rich text example, if accessed from android device the mark is not inserted on the inserted text.

**Recording**
![bug-mark](https://user-images.githubusercontent.com/16449020/122640473-62bc2700-d132-11eb-9102-0fd1f49d5223.gif)

**Sandbox**
It was replicated on https://www.slatejs.org/examples/richtext by accessing the site from android phone or chrome desktop with device set  to google pixel that emulate android

**Steps**
To reproduce the behavior:
1. https://www.slatejs.org/examples/richtext
2. toggle mark button
3. insert some text
4. text inserted without the mark

**Expectation**
mark should be inserted with the selected mark

**Environment**
- Slate Version: 
  - slate: 0.63.0
  - slate-react: 0.65.2
- Operating System: Android 9 - 11
- Browser: Chrome

**Context**
i found out [here](https://github.com/ianstormtaylor/slate/blob/main/packages/slate-react/src/components/android/android-input-manager.ts#L108), the mark is not checked when inserting text on android. im opening a PR #4342 but is there any concern why insert text is not reading the editor mark before?

**Comment by imdbsd** [comment]
if anybody facing the same issue as me. My current workaround was checking the editor.marks in onDOMBeforeInput. 
```
(event) => {
if (event.type === 'beforeinput' && event.inputType === 'insertText' &&  IS_ANDROID) {
      const text = event.data;
      const marks = editor.marks;
      if (typeof text === 'string' && marks) {
        event.preventDefault();
        const node = {text, ...marks};
        Transforms.insertNodes(editor, node, {
          match: Text.isText,
          at: editor.selection,
          select: true,
        });
        editor.marks = null;
      }
}
}
```

**Comment by imdbsd** [comment]
Closed, since #4342 was merged
